resources/shellHelperFunctions.sh

```diff
@@ -178,11 +178,54 @@ set_npm_version () {
   npm version --allow-same-version --no-git-tag-version --no-commit-hooks "$VERSION_WITH_TAG"
 }
 
+NUM_RE='^[0-9]+$'
+
+RETRY_MAX=5
+
+# Configuration:  wait between 10 sec and 120 sec.
+
+# in seconds.
+RETRY_MAX_WAIT_RANGE=111
+# in seconds
+RETRY_MIN_WAIT=10
+
+# $1  The current retry count (optional - defaults to 0 if not a non-negative whole number)
+# $2+ (everything else) the command to retry should it fail
+reattempt_if_failing ( ) {
+  local retryCount=$1
+
+  if ! [[ "$retryCount" =~ $NUM_RE ]]; then
+    retryCount=0
+  else
+    shift
+  fi
```

Comment:
This might be neater with a wrapper function, instead of a regex match on a parameter:

```
reattempt_if_failing() {
  _reattempt_if_failing 0 "$@"
}

_reattempt_if_failing() {
  ...
}
```